src/Client/ClickHouseClient.cpp

```diff
@@ -0,0 +1,286 @@
+#include <Common/NetException.h>
+#include <Client/ClickHouseClient.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int DEADLOCK_AVOIDED;
+extern const int TIMEOUT_EXCEEDED;
+extern const int UNKNOWN_PACKET_FROM_SERVER;
+extern const int UNEXPECTED_PACKET_FROM_SERVER;
+}
+
+namespace
+{
+
+std::unique_ptr<Connection> createConnection(const ConnectionParameters & parameters)
+{
+    auto ret = std::make_unique<Connection>(
```

Comment:
Let's leave a comment or file an issue for connection pooling. We can revisit `Connection` concept / pooling systematically for external stream / tables etc

Reply:
Switched to ConnectionPool.